addon/index.js

```diff
@@ -0,0 +1,52 @@
+import { decorator } from '@ember-decorators/utils/decorator-wrappers';
+import { run } from '@ember/runloop';
+
+function decoratorWithReturnValue(fn) {
+  return decorator(function(target, key, desc/*, params*/) {
+    // set as value on descriptor - a function that when called will 
+    // return run.next(myFunc)
+    return function(...args)  {
+      const { value } = desc;
+      return fn(this, value, ...args);
+    }
+  });
+}
+
+/**
+ * Decorator that makes the target function runloop aware
+ *
+ * ```js
+ * import Component from '@ember/component';
+ * import { next } from 'ember-decorators/runloop';
+ *
+ * export default class ActionDemoComponent extends Component {
+ *   @next
+ *   foo() {
+ *     // do something
+ *   }
+ * }
+ * ```
+ * @function
+ */
+export const next = decoratorWithReturnValue(run.next);
+
+/**
+ * Decorator that makes the target function runloop aware
+ * and bound to the context for which it exists
+ *
+ * ```js
+ * import Component from '@ember/component';
+ * import { bind } from 'ember-decorators/runloop';
+ *
+ * export default class ActionDemoComponent extends Component {
+ *   @bind
+ *   foo() {
+ *     // do something
+ *   }
+ * }
+ * ```
+ * @function
+ */
+export const bind = decorator((target, value, desc) => {
+  return run.bind(target, desc.value);
```

Comment:
This isn't working atm.  `target` is just an object...not the context of this.

Reply:
I think `bind` will have to use an initializer to set the function up when the class in instantiated. Let's do this in a separate followup PR